r/R/expression.R

```diff
@@ -28,5 +28,10 @@ runExpression <- function(req, data) {
 
   gene_subset <- gene_subset[, c("input", "name")]
 
-  return(getGeneExpression(data, gene_subset))
+  if (req$body$downsampled) {
+    return(getGeneExpression(data, gene_subset, req$body$cellIds))
```

Comment:
getGeneExpresson downsampled is alrady implemented (and used by the marker heatmap task), so we can just call it here instead of adding any other code.